### PR TITLE
Navigation: implement underlay

### LINKED ISSUE #47191: Expandable Sidemenu: implement backdrop overlay

### PR DESCRIPTION
**Pull request by ashharrison90**
**What this PR does / why we need it**:

# before
https://user-images.githubusercontent.com/20999846/161740149-df4da5a6-0825-4ec4-822a-cd6e209c508e.mp4 

# after
https://user-images.githubusercontent.com/20999846/161740195-526a72d7-1c58-49d7-86fd-01005b8b9fb5.mp4

**Which issue(s) this PR fixes**:

Fixes #47191 

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/core/components/NavBar/Next/NavBarMenu.tsx [hunk 1]
```diff
@@ -83,9 +102,99 @@ const getStyles = (theme: GrafanaTheme2) => ({
...
+
+  return {
+    underlay: {
+      appear: css({
+        ...underlayClosed,
+      }),
+      appearActive: css({
+        ...underlayTransition,
+        ...underlayOpen,
+      }),
```
Comment (joshhunt) [inline public/app/core/components/NavBar/Next/NavBarMenu.tsx:164]: can you pass these in as `css(underlayTransition, underlayOpen)`?

**Review by joshhunt — COMMENTED** [review]
it's an unimportant nit, but im not sure how I feel about the name "underlay" - I get especially confused when it's contrasted with "overlay". "backdrop" seems more standard, no?

Also, the colour doesnt seem correct? The Query Inspector uses a darker color.

1528_2022-04-05-13-51_chrome

**Comment by ashharrison90** [comment]
@joshhunt it uses the exact same colour as the drawer/modal backdrop, it's just that [the colour got changed 2 weeks ago](https://github.com/grafana/grafana/pull/46630) 👍

File: public/app/core/components/NavBar/Next/NavBarMenu.tsx (hunk 1 above)
Comment (ashharrison90) [reply public/app/core/components/NavBar/Next/NavBarMenu.tsx]: yeah this is much nicer

**Comment by joshhunt** [comment]
oh lol haha. serves me right for checking on grafana cloud.

tbh im not a total fan, but there's bigger things to get worked up over.

**Review by joshhunt — APPROVED** [review]

**Pull request merged**